README.md

```diff
@@ -231,6 +251,28 @@ For example, if some node is tainted NoSchedule or fails to start a job for some
 
 [![asciicast](https://asciinema.org/a/11wir751y89617oemwnsgli4a.png)](https://asciinema.org/a/11wir751y89617oemwnsgli4a)
 
+### Configuration options
+
+NFD supports a configuration file. The default location is
+`/etc/kubernetes/node-feature-discovery/node-feature-discovery.conf`
+,but, this can be changed by
+specifying the`--config` command line flag. The file is read inside the Docker
+image. Thus, Volumes and VolumeMounts are needed, or, a custom Docker image
+must be built in order to specify your settings via a configuration file.
+The (empty) [example config](https://github.com/kubernetes-incubator/node-feature-discovery/blob/master/node-feature-discovery.conf.example)
+is available in the NFD Docker image, which can be used as base in custom-built
+images.
+
+Configuration options can also be specified via the `--options` command line
+flag, in which case no mounts need to be used. The same format as in the config
+file must be used, i.e. YAML or JSON.
```

Comment:
Command line argument in yaml format sounds like a bad idea to me in terms of usability and user experience, imagine how confusing would be probably doing something this:
```
/usr/bin/node-feature-discovery --config="sources:\
  kernel:\
    kconfigFile: \"/path/to/kconfig\"\
    configOpts:\
      - \"NO_HZ\"\
      - \"X86\"\
      - \"DMI\"" --another-options="blah-blah" ...
```
especially that one option is to pass this as a yaml, where you have to take care of number of spaces, indentations, etc.
Now imagine something even more crazy: setting this option inside pod specification, where you would end up with yaml config in bash command in yaml config :)

I understand that the purpose of this command line argument is to be able to quickly test a new configuration and it's great to have this options, but I just wanted to point out that there may be some problems on the user side.

Reply:
Well, the idea of course is for people to use json. But using ghodss/yaml gives you both json and yaml so...

Reply:
isn't json as ugly as yaml in this scenario? :wink: but as I said earlier it's always better to have an option. Also, I can't think of any significantly better alternative, except maybe instead of `--config=<yaml_or_json>` split it to `--kernel-config-file=<filepath> --kernel-config-opts=NO_HZ,X86,DMI,...`. Not sure how it's gonna look in the future tho, if more plugins will decide to add their own kernel flags.

Reply:
Yeah, I thought something like this but forgot the idea because it makes everything more complex, it isn't future proof and expressive enough to cover all scenarios. JSON on the command line is not beatiful but I think this is is sometimes a handy way specifying one single config option, for example.